bazel/foreign_cc/BUILD

```diff
@@ -150,7 +150,6 @@ configure_make(
         "ICU_DATA_FILTER_FILE": "$(execpath //bazel/foreign_cc:icu_data_filter.json)",
     },
     lib_source = "@com_github_unicode_org_icu//:all",
-    linkopts = ["-lstdc++"],
```

Comment:
wonder if this should have this in the CXXFLAGs as in:

`CXXFLAGS": "-lstdc++` used for ragel. That seems to have worked in this case: https://github.com/envoyproxy/envoy/pull/19473#discussion_r790427535 using the `ENVOY_STDLIB` override.

Will defer to the expert (Lizan) in this area.

Reply:
I think if the `configure_make` does not produce any binaries like ragel, then it is safe to remove `-lstdc++`.

Reply:
@zhxie It is true that there won't be any other binaries generated besides envoy-static, so it is safe to remove `-lstdc++`.